app/models/manageiq/providers/oracle_cloud/inventory/collector/cloud_manager.rb

```diff
@@ -1,12 +1,40 @@
 class ManageIQ::Providers::OracleCloud::Inventory::Collector::CloudManager < ManageIQ::Providers::OracleCloud::Inventory::Collector
-  def connection
-    @connection ||= manager.connect
+  def compartments
+    @compartments ||= begin
+      root_compartment = identity_client.get_compartment(identity_client.api_client.config.tenancy).data
+
+      compartments = [root_compartment]
+      compartments += identity_client.list_compartments(root_compartment.id, :access_level => "ANY", :compartment_id_in_subtree => true).data
```

Comment:
Super minor, but it might be cleaner to use `.unshift` to prepend the root compartment.

Reply:
:+1: done